Marsiple/App/Todos/TodosViewController.swift

```diff
@@ -101,52 +105,40 @@ private extension TodosViewController {
-            var reloadTableView: (Bool) -> Void
-            let rowsToReload = strongSelf.indexPathsAfter(indexPath: indexPath)
-            if indexPath.section == 0 {
-                reloadTableView = { [weak self] _ in
-                    self?.todosView.tableView.reloadSections([finishIndexPath.section], with: .automatic)
-                    self?.todosView.tableView.reloadRows(at: rowsToReload, with: .automatic)
-                }
-            } else {
-                reloadTableView = { [weak self] _ in
-                    self?.todosView.tableView.reloadRows(at: rowsToReload, with: .automatic)
-                    self?.todosView.tableView.reloadRows(at: [finishIndexPath], with: .automatic)
-                }
-            }
-            
-            let removedTodo = strongSelf.todos[indexPath.section].remove(at: indexPath.row)
-            removedTodo.completed = !removedTodo.completed
-            strongSelf.todos[finishIndexPath.section].insert(removedTodo, at: finishIndexPath.row)
-            strongSelf.todosView.tableView.performBatchUpdates({
-                strongSelf.todosView.tableView.moveRow(at: indexPath, to: finishIndexPath)
-            }, completion: reloadTableView)
-        }
-        return didSelectCheckBox
+    func didSelectCheckBox(atIndexPath indexPath: IndexPath) {
+        // TODO: Save changes on the server
+        let finishIndexPath = IndexPath(row: indexPath.section * todos[1 - indexPath.section].count, section: 1 - indexPath.section)
+        moveTodo(at: indexPath, to: finishIndexPath)
+        moveRow(at: indexPath, to: finishIndexPath)
     }
     
-    func indexPathsAfter(indexPath: IndexPath) -> [IndexPath] {
-        var indexPaths = [IndexPath]()
-        let maxIndexOfTodos = todos[indexPath.section].count - 2
-        guard indexPath.row <= maxIndexOfTodos else {
-            return []
-        }
-        for i in indexPath.row...maxIndexOfTodos {
-            indexPaths.append(IndexPath(row: i, section: indexPath.section))
-        }
-        return indexPaths
+    func moveTodo(at startIndexPath: IndexPath, to finishIndexPath: IndexPath) {
+        var removedTodo = todos[startIndexPath.section].remove(at: startIndexPath.row)
+        removedTodo.completed = !removedTodo.completed
+        todos[finishIndexPath.section].insert(removedTodo, at: finishIndexPath.row)
     }
     
-    func didChangeText(atIndexPath indexPath: IndexPath) -> (() -> Void)? {
-        return { [weak self] in
-            self?.todosView.tableView.reloadRows(at: [indexPath], with: .automatic)
+    func moveRow(at startIndexPath: IndexPath, to finishIndexPath: IndexPath) {
+        todosView.tableView.performBatchUpdates({
+            todosView.tableView.moveRow(at: startIndexPath, to: finishIndexPath)
+        }, completion: { [weak self] _ in
+            self?.todosView.tableView.reloadData()
```

Comment:
I would like to see the `cell` that was `moved` `reloaded` right after the `move` completes. Finally, after the `reload` completes, we need to `reload the whole table view`. 

Hint: This will be messy a bit, since you will need to nest multiple `tableView.performBatchUpdates` calls.

Reply:
Why do I need to nest multiple `tableView.performBatchUpdates`? I believe that we would get the same result if `self?.todosView.tableView.reloadRows(at: [finishIndexPath], with: .automatic)` was added before `self?.todosView.tableView.reloadData()`, just like it used to be.

Reply:
You're right, we can do that.